router/base-http/src/main/scala/io/buoyant/router/http/MaxHopsFilter.scala

```diff
@@ -0,0 +1,47 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.Fields.Via
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.util.Future
+import io.buoyant.router.http.ForwardClientCertFilter.Enabled
+import io.buoyant.router.http.MaxHopsFilter.MaxHopsExceeded
+import scala.util.control.NoStackTrace
+
+class MaxHopsFilter[Req, H: HeadersLike, Rep](maxHops: Int, headerKey: String)(implicit requestLike: RequestLike[Req, H]) extends SimpleFilter[Req, Rep] {
+
+  def numHops(viaValue: String) = viaValue.split(",").length
+
+  def apply(req: Req, svc: Service[Req, Rep]): Future[Rep] = {
+    val headersLike = implicitly[HeadersLike[H]]
+
+    headersLike.get(requestLike.headers(req), headerKey) match {
+      case Some(v) if numHops(v) > maxHops => throw MaxHopsExceeded(maxHops)
+      case _ => svc(req)
+    }
+  }
+}
+
+object MaxHopsFilter {
+
+  final case class MaxHopsExceeded(numHops: Int)
+    extends Exception(s"Maximum number of hops ($numHops) has been exceeded. Please check for proxy loops.")
+    with NoStackTrace
+
+  final case class Param(value: Int) extends AnyVal {
+    def mk(): (Param, Stack.Param[Param]) = (this, Param.param)
+  }
+
+  object Param {
+    implicit val param = Stack.Param(Param(10))
```

Comment:
With a pure Linkerd self loop, we currently get to 1226 hops before hitting the `431 Request Header Fields Too Large`.  Let's set this default value larger to more closely match the behavior today and not break existing apps which have a call depth greater than 10.  What about 1000?